src/main/java/kafdrop/config/KafkaConfiguration.java

```diff
@@ -28,10 +28,13 @@ public void applyCommon(Properties properties) {
     properties.setProperty(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, brokerConnect);
     if (isSecured) {
       LOG.warn("The 'isSecured' property is deprecated; consult README.md on the preferred way to configure security");
-      properties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, securityProtocol);
       properties.put(SaslConfigs.SASL_MECHANISM, saslMechanism);
     }
 
+    if(isSecured || securityProtocol.equals("SSL")) {
```

Comment:
I don't understand why this is required only for SSL.
From [doc](https://kafka.apache.org/24/javadoc/org/apache/kafka/common/security/auth/SecurityProtocol.html) security protocol can be:
- PLAINTEXT
- SASL_PLAINTEXT
- SASL_SSL
- SSL

Why here we just check SSL?

Reply:
cc @Bert-R

Reply:
we are checking for isSecured or SSL and not just for SSL.
isSecured already covers for SASL. what remains is that we apply the security configuration in case of just SSL. This is what is done.  otherwise mTLS was only working for SSL_SASL and not just SSL.

we don't need security configuration for PLAINTEXT.

Reply:
One more question: Why `isSecured` is false in case of `SSL`?

P.S. Sorry to ask but I'm not an expert, just trying to maintain the project ...

Reply:
The ``isSecured`` property is old. It's deprecated since 2019, see line 30. So that should generally not be set to ``true``.

Reply:
That would break the legacy `isSecured` feature, as the property put is removed from line 31.

Reply:
isSecured is pulled from the application config, look at kafdrop.java:127 and application.yaml:35. so in case someone does set it true. It needs to work, which is why I kept it as is.  We can raise a separate issue to remove the whole isSecured part and then clean it up.